services/rds/src/main/java/org/finra/gatekeeper/services/group/service/GatekeeperLdapRoleLookupService.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2018. Gatekeeper Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.finra.gatekeeper.services.group.service;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.finra.gatekeeper.common.properties.GatekeeperAuthProperties;
+import org.finra.gatekeeper.common.services.account.AccountInformationService;
+import org.finra.gatekeeper.common.services.user.auth.GatekeeperAuthorizationService;
+import org.finra.gatekeeper.common.services.user.search.GatekeeperLdapLookupService;
+import org.finra.gatekeeper.configuration.GatekeeperRdsAuthProperties;
+import org.finra.gatekeeper.services.group.interfaces.IGatekeeperRoleLookupService;
+import org.finra.gatekeeper.services.group.model.GatekeeperADGroupEntry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ldap.core.AttributesMapper;
+import org.springframework.ldap.core.LdapTemplate;
+import org.springframework.ldap.query.LdapQueryBuilder;
+import org.springframework.stereotype.Component;
+
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+
+@Component
+public class GatekeeperLdapRoleLookupService implements IGatekeeperRoleLookupService {
```

Comment:
We can levarage the existing GatekeeperRoleService to implement pulling down the user's authorized requestor groups.